### PR TITLE
add jobs for expiring snowflake passwords

### PR DESCRIPTION
**Pull request by estute**
This adds the DSL to create 2 jobs that run the `expire_user_passwords.py` script defined in https://github.com/edx/analytics-tools/pull/25:
* expire-snowflake-passwords-quarterly (an automated job)
* expire-snowflake-passwords-manually (a manual job for fixing issues when the first job fails)

### CODE REVIEW

File: dataeng/jobs/analytics/SnowflakeExpirePasswords.groovy [hunk 1]
```diff
@@ -0,0 +1,88 @@
+package analytics
+
+import static org.edx.jenkins.dsl.AnalyticsConstants.common_multiscm
+import static org.edx.jenkins.dsl.AnalyticsConstants.common_parameters
```
Comment (HassanJaveed84) [inline dataeng/jobs/analytics/SnowflakeExpirePasswords.groovy]: Unused.

File: dataeng/jobs/analytics/SnowflakeExpirePasswords.groovy [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+import static org.edx.jenkins.dsl.AnalyticsConstants.common_log_rotator
+import static org.edx.jenkins.dsl.AnalyticsConstants.common_wrappers
```
Comment (HassanJaveed84) [inline dataeng/jobs/analytics/SnowflakeExpirePasswords.groovy]: unused.

File: dataeng/jobs/analytics/SnowflakeExpirePasswords.groovy (hunk 1 above)
Comment (HassanJaveed84) [inline dataeng/jobs/analytics/SnowflakeExpirePasswords.groovy]: unused

File: dataeng/jobs/analytics/SnowflakeExpirePasswords.groovy [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+        jobConfigs.each { jobConfig ->
+
+            dslFactory.job(jobConfig['NAME']) {
+
+                logRotator common_log_rotator(allVars)
+                parameters secure_scm_parameters(allVars)
+                parameters {
+                    stringParam('ANALYTICS_TOOLS_URL', allVars.get('ANALYTICS_TOOLS_URL'), 'URL for the analytics tools repo.')
+                    stringParam('ANALYTICS_TOOLS_BRANCH', allVars.get('ANALYTICS_TOOLS_BRANCH'), 'Branch of analtyics tools repo to use.')
```
Comment (brianhw) [inline dataeng/jobs/analytics/SnowflakeExpirePasswords.groovy]: nit:   analtyics => analytics

File: dataeng/jobs/analytics/SnowflakeExpirePasswords.groovy [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+            SnowflakeExpirePasswordsQuarterlyConfig,
+            SnowflakeExpirePasswordsManuallyConfig
+        ]
+
+        jobConfigs.each { jobConfig ->
+
+            dslFactory.job(jobConfig['NAME']) {
+
+                logRotator common_log_rotator(allVars)
+                parameters secure_scm_parameters(allVars)
```
Comment (HassanJaveed84) [inline dataeng/jobs/analytics/SnowflakeExpirePasswords.groovy]: This job doesn't make use of secure scm parameters. We can get rid of `secure_scm_parameters` and `secure_scm` down below.

File: dataeng/jobs/analytics/SnowflakeExpirePasswords.groovy (hunk 1 above)
Comment (HassanJaveed84) [inline dataeng/jobs/analytics/SnowflakeExpirePasswords.groovy]: unused.

**Pull request merged**